pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -62,27 +63,27 @@ func (nameScoping *DefaultNameScopingImpl) Handle(command utils.CommandEnum, clu
 	case utils.CONTAINER_CREATE:
 		var newQuery string
 		var buf bytes.Buffer
-		var containerConfig dockerclient.ContainerConfig
+		var oldconfig clusterParams.OldContainerConfig
 		var reqBody []byte
 		defer r.Body.Close()
 		if reqBody, _ = ioutil.ReadAll(r.Body); len(reqBody) > 0 {
-			if err := json.NewDecoder(bytes.NewReader(reqBody)).Decode(&containerConfig); err != nil {
+			if err := json.NewDecoder(bytes.NewReader(reqBody)).Decode(&oldconfig); err != nil {
 				log.Error(err)
 				return err
 			}
 		}
 		if "" != r.URL.Query().Get("name") {
 			log.Debug("Postfixing name with tenantID...")
 			newQuery = strings.Replace(r.RequestURI, r.URL.Query().Get("name"), r.URL.Query().Get("name")+r.Header.Get(headers.AuthZTenantIdHeaderName), 1)
-			containerConfig.Labels[headers.OriginalNameLabel] = r.URL.Query().Get("name")
+			oldconfig.ContainerConfig.Config.Labels[headers.OriginalNameLabel] = r.URL.Query().Get("name")
 		}
-		containerConfig.HostConfig.NetworkMode = getNetworkID(cluster, r, containerConfig.HostConfig.NetworkMode)
-		if err := CheckContainerReferences(cluster, r.Header.Get(headers.AuthZTenantIdHeaderName), &containerConfig); err != nil {
+		oldconfig.ContainerConfig.HostConfig.NetworkMode = containertypes.NetworkMode(getNetworkID(cluster, r, string(oldconfig.ContainerConfig.HostConfig.NetworkMode)))
+		if err := CheckContainerReferences(cluster, r.Header.Get(headers.AuthZTenantIdHeaderName), &oldconfig); err != nil {
 			log.Error(err)
 			return err
 		}
 		if len(reqBody) > 0 {
-			if err := json.NewEncoder(&buf).Encode(containerConfig); err != nil {
+			if err := json.NewEncoder(&buf).Encode(oldconfig); err != nil {
 				log.Error(err)
```

Comment:
Is it always like that?
Return error after logging it?
It should have uniform behaviour otherwise we will print the same error multiple times..